source/coreml.js

```diff
@@ -441,11 +479,159 @@ coreml.Graph = class {
                     [ model.description.output[0].name ]);
                 return 'customModel';
             }
+            case 'mlProgram': {
+                return this._loadMLProgram(model.mlProgram, scope, group);
+            }
         }
         throw new coreml.Error("Unknown model type '" + JSON.stringify(Object.keys(model)) + "'.");
     }
 
-    _createNode(scope, group, type, name, description, data, inputs, outputs) {
+    static _formatMILOpType(opName) {
+        switch (opName) {
```

Comment:
Why is this translation done? Would not translating the names add more clarity? Which of these two options describes the differences better?

Option 1) All ML program operators are different from the `neuralNetwork` ones and have a different schema, even ones with the same name like `reshape`. The names should be prefixed and have a duplicate `coreml-metadata.json` entry keeping the underlying ML program name.

Option 2) Only ML program operators with a different name have a different schema. For example, `reshape` is identical. Keep the names (no prefix) and add metadata for new operators to  `coreml-metadata.json`.

Reply:
Good question. I had started translating names when I found that the MLProgram representation used a different operator name from the equivalent neuralNetwork layer, but my translations here are not comprehensive.

I'll remove this translation layer, and I think your Option 2 sounds like the right way to proceed -- we should have coverage in coreml-metadata.json for all the expected operator type names in MLProgram, in addition to the ones that already exist in neuralNetwork. I think we can assume that if an operation exists in both with the same name, that it refers to the same operation.

Reply:
I'm not sure if I understand "schema" correctly. 
But want to clarify:  even if an "op" in ML Program has the same name as a "layer" in neural network, e.g. `reshape`, it can different parameters/ different behavior.

Reply:
Thanks for clarifying @aseemw. I think in that case @lutzroeder's Option 1 makes the most sense.